internal/project/projectdata/packageindex.go

```diff
@@ -115,33 +116,44 @@ func getPackageIndexData(interfaceObject map[string]interface{}, pointerPrefix s
 	}
 
 	for index, interfaceElement := range interfaceSlice {
-		object, ok := interfaceElement.(map[string]interface{})
-		if !ok {
-			continue
+		interfaceElementData := PackageIndexData{
+			JSONPointer: fmt.Sprintf("%s/%s/%v", pointerPrefix, dataKey, index),
+			Object:      nil,
 		}
 
-		var iD string
-		iDSuffix, ok := object[iDKey].(string)
-		if !ok {
-			continue
+		object, ok := interfaceElement.(map[string]interface{})
+		if ok {
+			interfaceElementData.Object = object
 		}
-		iD = iDPrefix + iDSuffix
-		if versionKey != "" {
-			iDVersion, ok := object[versionKey].(string)
+
+		objectID := func() string {
+			if iDPrefix != "" && strings.HasPrefix(iDPrefix, pointerPrefix) {
```

Comment:
I think we can avoid checking that `iDPrefix` is not an empty string here. :thinking:

Reply:
It's necessary because it is an empty string when creating the packages slice:
https://github.com/arduino/arduino-lint/blob/901fcd2adef85d748d966ccfd64c7415a8397cff/internal/project/projectdata/packageindex.go#L45

I'm not super happy with this approach to detecting when the ID prefix is using the fallback ID, but I didn't come up with anything better.

Reply:
Aaah ok, fine for me then.